### DISCUSSION TITLE
GetDefinitionsProductType API Size attribute vulnerability

### DISCUSSION THREAD

**Discussion by wszwc** [amzn/selling-partner-api-models#3840] [General]
I have developed the json product type to the final stage, but there are still loopholes that cannot be solved. I am not sure whether it is a loophole or I have not found a solution. Please let the official staff help me:
1. I used allof validation in JSON Schema and I can easily get all the unfilled required fields based on the current data, but it seems that it cannot derive the constraints of the disabled fields.
For example, because the A attribute is filled in, the B attribute is disabled.
2. The theme size has different names in different product types, but they have the same internal attributes such as size_class, size_system, and size.
But there are constraints between the inner properties and I can still know the unfilled required fields from allof, which is fine, but what? How to obtain the enumeration value corresponding to each size_class?
The size enumeration value contains all the enumerations required by size_class. I have no way of knowing what the size enumeration corresponding to each size_class is?
When size_class is equal to age, the corresponding size enumeration value still contains enumeration values other than age.

**Comment by wszwc** [comment]
Now I can never successfully publish it because of the size field error. This is too bad. Even if I manually modify it in the background, the enumeration value is still as high as more than 300.
But when I create it directly in the background, the enumeration value is normal. This is unfair. Products uploaded through the interface are inherently stricter than the background itself, and there are loopholes.

**Comment by jackmin-sw** [comment]
I have the same issue, it's not easy to get the variation_theme value for the product, but when we download the product template, the variation_theme only have a few data.

**Reply by wszwc** [reply]
Hello, I have not encountered this problem. Did you set the requirements to LISTING when requesting the interface?
In addition, if you want to get the variant themes available for the current product type, you only need to remove the deprecated theme collection in the variant theme field.
And this is the type of problem I'm having:
1. I cannot know what attributes correspond to the variant theme, because the variant theme name is inconsistent with the attribute name. I have now solved this problem using similarity matching, but there is still a risk
2. Because there are mutual constraints between attributes, and verification can only be obtained based on existing data, which makes dynamic verification almost impossible to achieve.
For example: when the size system is age, the size attribute needs to fill in the age-related enumeration. Amazon provides size enumerations required by all size systems by default. If you want to obtain the size enumeration for the corresponding size system, you need to let the size attribute have a value before it is verified, not after selecting the size system. The enumeration is checked immediately, which makes verification difficult.
3. The verification is very imperfect, and there are often various unexpected situations and unreasonable places.
For example: when I set the battery required property to No, the validation doesn't tell me that the battery property shouldn't be filled in, which makes the validation almost unusable. Doesn't work on forms at all

**Reply by jackmin-sw** [reply]
@wszwc thanks，1. filter the deprecated value in the "enumDeprecated" property can solve the issue. 
2. The other issues you list, I can't find a solution either！
